testassets/FunctionalTestsWebsite/Greeter.cs

```diff
@@ -37,7 +38,38 @@ public override Task<HelloReply> SayHello(HelloRequest request, ServerCallContex
         return Task.FromResult(new HelloReply { Message = "Hello " + request.Name });
     }
 
-    public override async Task SayHellos(HelloRequest request, IServerStreamWriter<HelloReply> responseStream, ServerCallContext context)
+    public override Task SayHellos(HelloRequest request, IServerStreamWriter<HelloReply> responseStream, ServerCallContext context)
+    {
+        return SayHellosCore(request, responseStream);
+    }
+
+    public override Task SayHellosBufferHint(HelloRequest request, IServerStreamWriter<HelloReply> responseStream, ServerCallContext context)
+    {
+        context.WriteOptions = new WriteOptions(WriteFlags.BufferHint);
+
+        return SayHellosCore(request, responseStream);
+    }
+
+    public override Task SayHellosSendHeadersFirst(HelloRequest request, IServerStreamWriter<HelloReply> responseStream, ServerCallContext context)
+    {
+        context.WriteResponseHeadersAsync(null);
+
+        return SayHellosCore(request, responseStream);
+    }
+
+    public override async Task SayHellosDeadline(HelloRequest request, IServerStreamWriter<HelloReply> responseStream, ServerCallContext context)
+    {
+        var i = 0;
+        while (DateTime.UtcNow < context.Deadline)
```

Comment:
rather than watching the current time, we should probably watch the context.CancellationToken (and framework should cancel it when the deadline is crossed).

Reply:
I've added a test for that as well.